tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -1286,3 +1291,65 @@ def test_supported_bucket_policy_operations(self, mcg_obj, bucket_factory):
         assert (
             not missing_policies
         ), f"Some bucket_policies are not created : {missing_policies}"
+
+
+@polarion_id("OCS-6540")
+class TestNoobaaUpgradeWithBucketPolicy:
+    """
+    Test noobaa status post upgrade when there is bucket
+    with some bucket policy
+
+    """
+
+    @pre_upgrade
+    def test_create_bucket_policy_before_upgrade(
+        self,
+        request,
+        bucket_factory_session,
+        mcg_obj_session,
+    ):
+        """
+        Create bucket with some bucket policy before the upgrade
+
+        """
+        # Create a bucket and create obc object
+        obc_name = bucket_factory_session(amount=1, interface="OC")[0].name
+        obc_obj = OBC(obc_name)
+
+        # Generate bucket policy
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=["PutBucketPolicy", "GetBucketPolicy", "DeleteBucketPolicy"],
+            resources_list=[obc_obj.bucket_name],
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(
+            "Caching the bucket and bucket policy info for post upgrade verification"
+        )
+        request.config.cache.set("bucket_policy_bucket", obc_name)
+        request.config.cache.set("bucket_policy", bucket_policy)
+
+    @post_upgrade
+    def test_verify_noobaa_after_upgrade(self, request, mcg_obj_session):
+        """
+        Verify the noobaa health and verify the bucket policy post upgrade
+
+        """
+        logger.info("Extracting the bucket and bucket policy info from the cache")
+        obc_name = request.config.cache.get("bucket_policy_bucket", None)
+        bucket_policy = request.config.cache.get("bucket_policy", None)
+        assert (
+            obc_name and bucket_policy
+        ), "Seem like either pre-upgrade test for this failed or unable to cache the bucket/bucket policy info"
+
+        # Check noobaa health
+        logger.info("Verifying noobaa health")
+        CephCluster().noobaa_health_check()
+
+        # Check backingstore health
+        verify_backing_store()
```

Comment:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/storage_cluster.py/#L1323-L1334

Might be a misunderstanding on my part, but this function doesn't seem to query the status of a specific backingstore. Couldn't this fail the test if we have an unrelated backingstore leftover in a bad state?

Reply:
You are right, I think i need fix this